src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -32,7 +32,7 @@
   protected static final String DEFAULT_CHANCE = "1:1";
   protected static final String CHANCE = "chance";
   // list of conditions that this condition can
-  protected ArrayList<RulesAttachment> m_conditions = new ArrayList<>();
+  protected List<RulesAttachment> m_conditions = new ArrayList<>();
```

Comment:
Is the code in `ValidateAttachmentsTest#validateAttachment()` the only constraint that's been forcing us to use concrete types for attachment fields and accessors?

Reply:
No idea, seems like it. Seems to work.

Reply:
@ron-murhammer Do you have any history here?  Just want to make sure there's nothing subtle we're missing.  It seems strange that someone would have gone to this much trouble to use concrete types everywhere if it was only self-imposed by a test.

Reply:
Unfortunately, I don't.